### ISSUE TITLE
Bug: StopIteration error in histplot

### ISSUE THREAD

**Issue by yannikschaelte** [mwaskom/seaborn#3072]
```python
import seaborn as sns

penguins = sns.load_dataset("penguins")
sns.histplot(data=penguins, x="flipper_length_mm")
```
gives a `StopIteration` error with matplotlib 3.6.1 (works with 3.6.0). seaborn version 0.12.0, python version 3.10.

Detailed error message:

```
---------------------------------------------------------------------------
StopIteration                             Traceback (most recent call last)
Input In [2], in <cell line: 2>()
      1 penguins = sns.load_dataset("penguins")
----> 2 sns.histplot(data=penguins, x="flipper_length_mm")

File ~/env/env/lib/python3.10/site-packages/seaborn/distributions.py:1418, in histplot(data, x, y, hue, weights, stat, bins, binwidth, binrange, discrete, cumulative, common_bins, common_norm, multiple, element, fill, shrink, kde, kde_kws, line_kws, thresh, pthresh, pmax, cbar, cbar_ax, cbar_kws, palette, hue_order, hue_norm, color, log_scale, legend, ax, **kwargs)
   1416     else:
   1417         method = ax.plot
-> 1418     color = _default_color(method, hue, color, kwargs)
   1420 if not p.has_xy_data:
   1421     return ax

File ~/env/env/lib/python3.10/site-packages/seaborn/utils.py:139, in _default_color(method, hue, color, kws)
    134     scout.remove()
    136 elif method.__name__ == "bar":
    137 
    138     # bar() needs masked, not empty data, to generate a patch
--> 139     scout, = method([np.nan], [np.nan], **kws)
    140     color = to_rgb(scout.get_facecolor())
    141     scout.remove()

File ~/env/env/lib/python3.10/site-packages/matplotlib/__init__.py:1423, in _preprocess_data.<locals>.inner(ax, data, *args, **kwargs)
   1420 @functools.wraps(func)
   1421 def inner(ax, *args, data=None, **kwargs):
   1422     if data is None:
-> 1423         return func(ax, *map(sanitize_sequence, args), **kwargs)
   1425     bound = new_sig.bind(ax, *args, **kwargs)
   1426     auto_label = (bound.arguments.get(label_namer)
   1427                   or bound.kwargs.get(label_namer))

File ~/env/env/lib/python3.10/site-packages/matplotlib/axes/_axes.py:2373, in Axes.bar(self, x, height, width, bottom, align, **kwargs)
   2371 x0 = x
   2372 x = np.asarray(self.convert_xunits(x))
-> 2373 width = self._convert_dx(width, x0, x, self.convert_xunits)
   2374 if xerr is not None:
   2375     xerr = self._convert_dx(xerr, x0, x, self.convert_xunits)

File ~/env/env/lib/python3.10/site-packages/matplotlib/axes/_axes.py:2182, in Axes._convert_dx(dx, x0, xconv, convert)
   2170 try:
   2171     # attempt to add the width to x0; this works for
   2172     # datetime+timedelta, for instance
   (...)
   2179     # removes the units from unit packages like `pint` that
   2180     # wrap numpy arrays.
   2181     try:
-> 2182         x0 = cbook._safe_first_finite(x0)
   2183     except (TypeError, IndexError, KeyError):
   2184         pass

File ~/env/env/lib/python3.10/site-packages/matplotlib/cbook/__init__.py:1749, in _safe_first_finite(obj, skip_nonfinite)
   1746     raise RuntimeError("matplotlib does not "
   1747                        "support generators as input")
   1748 else:
-> 1749     return next(val for val in obj if safe_isfinite(val))

StopIteration: 
```

**Comment by mwaskom** [comment]
Hi, this is a bug in matplotlib 3.6.1: https://github.com/matplotlib/matplotlib/issues/24127

The only workaround for now is to downgrade to 3.6.0.

**Comment by jjerphan** [comment]
https://github.com/matplotlib/matplotlib/pull/24149 has resolved https://github.com/matplotlib/matplotlib/issues/24127 and has resolved the problem on my machine when [using the nightly build of matplotlib](https://matplotlib.org/stable/users/installing/index.html#installing-a-nightly-build) (`3.7.0.dev376+g23730b41fe` in my case).

**Comment by peterdsharpe** [comment]
@mwaskom While this may be a bug in matplotlib, this alone is an insufficient answer.

If seaborn requires `matplotlib <= 3.6.0` to work, and seaborn does not pin `matplotlib <= 3.6.0`, **that constitutes a bug in seaborn** that should be hotfixed.

**Comment by mwaskom** [comment]
Seaborn 0.11.2 was released with `matplotlib != 3.6.1` and matplotlib 3.6.2 is now out with the seaborn test suite passing on it, so I'm going to close this issue.
